src/gas/GasFeeController.ts

```diff
@@ -309,56 +355,61 @@ export class GasFeeController extends BaseController<
     this.fetchLegacyGasPriceEstimates = fetchLegacyGasPriceEstimates;
-    this.pollTokens = new Set();
+    this.fetchFeeHistory = fetchFeeHistory;
+    this.pollQueue = new Set();
+    this.pollableItems = {
+      gasFeeEstimates: this._fetchGasFeeEstimateData.bind(this),
+      isNetworkCongested: this.determineWhetherNetworkCongested.bind(this),
+    };
     this.getCurrentNetworkEIP1559Compatibility = getCurrentNetworkEIP1559Compatibility;
     this.getCurrentNetworkLegacyGasAPICompatibility = getCurrentNetworkLegacyGasAPICompatibility;
     this.getCurrentAccountEIP1559Compatibility = getCurrentAccountEIP1559Compatibility;
     this.EIP1559APIEndpoint = EIP1559APIEndpoint;
     this.legacyAPIEndpoint = legacyAPIEndpoint;
-    this.getChainId = getChainId;
-    this.currentChainId = this.getChainId();
+    this.getChainId = withNormalizedChainId(getChainId);
     const provider = getProvider();
     this.ethQuery = new EthQuery(provider);
     this.clientId = clientId;
     onNetworkStateChange(async () => {
       const newProvider = getProvider();
-      const newChainId = this.getChainId();
       this.ethQuery = new EthQuery(newProvider);
-      if (this.currentChainId !== newChainId) {
-        this.currentChainId = newChainId;
-        await this.resetPolling();
-      }
     });
   }
 
-  async resetPolling() {
-    if (this.pollTokens.size !== 0) {
-      const tokens = Array.from(this.pollTokens);
-      this.stopPolling();
-      await this.getGasFeeEstimatesAndStartPolling(tokens[0]);
-      tokens.slice(1).forEach((token) => {
-        this.pollTokens.add(token);
-      });
-    }
-  }
-
   async fetchGasFeeEstimates(options?: FetchGasFeeEstimateOptions) {
     return await this._fetchGasFeeEstimateData(options);
   }
 
+  /**
+   * Makes a request for gas fee estimates and adds that request to the
+   * polling queue. The polling queue is then started if it is not already
+   * started.
+   *
+   * @param _pollingToken - A polling token (unused).
+   * @returns The string "gasFeeEstimates".
+   * @deprecated
+   * @see updateWithAndStartPollingFor
+   */
   async getGasFeeEstimatesAndStartPolling(
-    pollToken: string | undefined,
-  ): Promise<string> {
-    const _pollToken = pollToken || random();
-
-    this.pollTokens.add(_pollToken);
+    _pollingToken: string | undefined,
+  ): Promise<keyof PollableItems> {
+    await this.updateWithAndStartPollingFor('gasFeeEstimates');
+    return 'gasFeeEstimates';
+  }
 
-    if (this.pollTokens.size === 1) {
-      await this._fetchGasFeeEstimateData();
-      this._poll();
+  /**
+   * Makes a request for the given item, updating the state internally with the resulting data, then
+   * adds the request to the polling queue. The polling timer is then started if it is not already
+   * started.
+   *
+   * @param item - The known name of a piece of data for which we are interested in polling.
+   */
+  async updateWithAndStartPollingFor(item: keyof PollableItems): Promise<void> {
```

Comment:
These changes remove the concept of a polling token, but on second thought, we should actually keep that. In rooting around in the `extension` repo, I've recently learned that we start duplicate instances of polling for gas fee estimates in multiple places. We shouldn't be doing this, but we do. So by no longer keeping track of polling tokens, we will make it so that if a component stops polling for gas fee estimates it kills polling for ALL instances.